### ISSUE TITLE
Proposer compute state root fails

### ISSUE THREAD

**Issue by terencechain** [prysmaticlabs/prysm#1977]
On the latest master. None of the proposer can propose due to the following:

```
ERROR validator: Not proposing! Unable to compute state root: rpc error: code = Unknown desc = could not execute state transition could not process block: could not process block attestations: could not verify attestation at index 0 in block: expected JustifiedBlockRoot == getBlockRoot(state, JustifiedEpoch)
```

**Comment by terenc3t** [comment]
Just saw this error again in run time:
`
"could not execute state transition could not process block: could not process block attestations: could not verify attestation at index 0 in block: expected attestation.JustifiedEpoch == state.JustifiedEpoch, received 32 == 31"
`

**Comment by prestonvanloon** [comment]
This functionality is currently enabled by flag only, so we have the flag turned off until we can resolve this issue. 

We should use the state replay tool to identify the root cause of this with some existing set of data. It seems to happen intermittently as the blockchain can still advance, but there are constant skip blocks and reorg conditions.

**Comment by SjonHortensius** [comment]
Even without an active validator - my beacon node just failed with the same error:

```
  INFO blockchain: State transition complete currentEpoch=17447 slotNumber=139583
  INFO blockchain: Finished processing beacon block slot=139583
  INFO blockchain: Applying LMD-GHOST Fork Choice Rule
  INFO blockchain: Chain head block and state updated headRoot=0x8c2018b227b9273950e0667a headSlot=139583 stateSlot=139583
  INFO blockchain: Executing state transition slotNumber=139584
 ERROR regular-sync: Could not process beacon block: block failed processing: could not process block: could not process block attestations: could not verify attestation at index 0 in block: expected attestation.JustifiedEpoch == state.JustifiedEpoch, received 17447 == 17446
  INFO blockchain: Executing state transition slotNumber=139584
 ERROR regular-sync: Could not process beacon block: block failed processing: could not process block: could not process block attestations: could not verify attestation at index 0 in block: expected attestation.JustifiedEpoch == state.JustifiedEpoch, received 17447 == 17446
```

after this it hangs and stops processing anything until restarted. I'm running  7c47db0 without any configuration - so I'd expect this wouldn't happen

**Comment by Doppelbock42** [comment]
I got a similar error:
```
[2019-05-11 15:09:12]  INFO blockchain: Slot transition successfully processed slotsSinceGenesis=169874
[2019-05-11 15:09:12]  INFO blockchain: Slot transition successfully processed slotsSinceGenesis=169875
[2019-05-11 15:09:12]  INFO blockchain: Slot transition successfully processed slotsSinceGenesis=169876
[2019-05-11 15:09:12]  WARN blockchain: Processed 45 skipped slots
[2019-05-11 15:09:12] ERROR regular-sync: Could not process beacon block: block failed processing: could not process block: could not process block attestations: could not verify attestation at index 0 in block: expected attestation.JustifiedRoot == state.PreviousJustifiedRoot, received 0x051e091bf74d4912c20b8fd2ea0f84398a362eee45a2c1f40b2d9d039466c1f7 == 0xb1a5f8191d600573ff2d8be32b100a491ab8a71586dbcaee3695a7183151abd1
[2019-05-11 15:13:03] ERROR attestation: Bitfield points to an invalid index in the committee: bitfield [00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00100000]
[2019-05-11 15:13:15] ERROR attestation: Bitfield points to an invalid index in the committee: bitfield [00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00100000]
```

**Comment by prestonvanloon** [comment]
This is resolved with latest update to 0.8.1. #2312
